dhis2/dhis2-core
# fix: remove OU from category options when OU is deleted [DHIS2-13152]

### Summary
Unlinks OUs and category options in case a OU is deleted.

### Automatic Testing
Added a new unit tests where a OU with CO is added and deleted

### Manual Testing
* create a new OU
* edit an existing CO and check the newly created OU as linked
* delete the new OU
* confirm the OU was deleted successful and the CO is no longer associated with the OU
